### PR TITLE
[Perpetual]: Refactor funding rate and interest calculation to use time

### PR DESCRIPTION
**Pull request by amityadav0**
## Description

- Refactor funding rate and interest calculation to use time

### What has Changed?

What specific problem were you aiming to address, and how did you successfully resolve it? If tests were not uploaded for this pull request or if coverage decreased, please provide an explanation for the change.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

**Comment by cosmic-vagabond** [comment]
@amityadav0 there is a conflict in this PR thanks

**Comment by amityadav0** [comment]
@cosmic-vagabond merged

File: x/perpetual/spec/09_funding_payment.md [hunk 1]
```diff
@@ -42,8 +42,8 @@ This formula ensures that the funding rate stays within defined boundaries, ince
 
 ### 4. **Funding Fee Collection**
 
-At every block, the calculated funding fees are collected and sent to the **Funding Fee Collection Address**. The details of this process can be found in the source code:  
-[Funding Fee Collection Code](https://github.com/elys-network/elys/blob/main/x/perpetual/keeper/handle_funding_fee_collection.go).
+At every block, the calculated funding fees are tracked. If a user makes a interaction(close position or consolidate) the fee collection is triggered and The details of this process can be found in the source code:
```
Comment (cosmic-vagabond) [inline x/perpetual/spec/09_funding_payment.md:45]: @amityadav0 "fee collection is triggered and…" the sentence seems to missing some words

File: x/perpetual/spec/09_funding_payment.md [hunk 2]
```diff
@@ -53,26 +53,31 @@ The distribution of funding fees is based on the calculated funding rate and ser
...
+However, the **actual distribution amount** is not directly proportional to the funding rate alone. Instead, the amount that each position receives is calculated based on the size of that position relative to the total liabilities/custody on the side that is receiving the funding payments.
 
 The formula for determining the amount distributed to each position is as follows:
 
 ```
 payment_to_position = (position_liabilities / pool_liabilities) * total_funding_collected
 ```
+```
+payment_to_position = (position_custody / pool_custody) * total_funding_collected
+```
```
Comment (cosmic-vagabond) [inline x/perpetual/spec/09_funding_payment.md:68]: @amityadav0 could you clarify here that one formula is for long and the other is for short positions thanks

File: x/perpetual/spec/09_funding_payment.md [hunk 3]
```diff
@@ -42,8 +42,8 @@ This formula ensures that the funding rate stays within defined boundaries, ince
 
 ### 4. **Funding Fee Collection**
 
-At every block, the calculated funding fees are collected and sent to the **Funding Fee Collection Address**. The details of this process can be found in the source code:  
-[Funding Fee Collection Code](https://github.com/elys-network/elys/blob/main/x/perpetual/keeper/handle_funding_fee_collection.go).
+At each block, the system tracks the calculated funding fees. If a user performs an interaction, such as closing a position or consolidating, the system triggers the collection of the accumulated fees and The details of this process can be found in the source code:
```
Comment (amityadav0) [inline x/perpetual/spec/09_funding_payment.md:45]:
```suggestion
At each block, the system tracks the calculated funding fees. If a user performs an interaction, such as closing a position or consolidating, the system triggers the collection of the accumulated fees. The details of this process can be found in the source code:  
```

**Review by cosmic-vagabond — APPROVED** [review]
LGTM

**Pull request merged**